sphinx/domains/rst.py

```diff
@@ -139,19 +139,21 @@ def handle_signature(self, sig, signode):
 
     def add_target_and_index(self, name, sig, signode):
         # type: (str, str, addnodes.desc_signature) -> None
+        directive_name = self.current_directive
         targetname = '-'.join([self.objtype, self.current_directive, name])
         if targetname not in self.state.document.ids:
             signode['names'].append(targetname)
             signode['ids'].append(targetname)
             signode['first'] = (not self.names)
             self.state.document.note_explicit_target(signode)
 
+            objname = ':'.join(filter(None, [directive_name, name]))
             domain = cast(ReSTDomain, self.env.get_domain('rst'))
-            domain.note_object(self.objtype, name, location=(self.env.docname, self.lineno))
+            domain.note_object(self.objtype, objname, location=(self.env.docname, self.lineno))
```

Comment:
I fixed here to register directive option with directive name to make them unique.